diff -Nru a/src/apps/ocioconvert/main.cpp b/src/apps/ocioconvert/main.cpp
--- a/src/apps/ocioconvert/main.cpp	2018-01-05 04:38:27.000000000 +0300
+++ b/src/apps/ocioconvert/main.cpp	2018-12-25 00:45:18.664364887 +0300
@@ -119,7 +119,7 @@
     std::cerr << "Loading " << inputimage << std::endl;
     try
     {
-        OIIO::ImageInput* f = OIIO::ImageInput::create(inputimage);
+        OIIO::ImageInput::unique_ptr f = OIIO::ImageInput::create(inputimage);
         if(!f)
         {
             std::cerr << "Could not create image input." << std::endl;
@@ -143,7 +143,7 @@
         memset(&img[0], 0, imgwidth*imgheight*components*sizeof(float));
         
         f->read_image(OIIO::TypeDesc::TypeFloat, &img[0]);
-        delete f;
+        f->close();
         
         std::vector<int> kchannels;
         //parse --ch argument
@@ -308,7 +308,7 @@
     // Write out the result
     try
     {
-        OIIO::ImageOutput* f = OIIO::ImageOutput::create(outputimage);
+        OIIO::ImageOutput::unique_ptr f = OIIO::ImageOutput::create(outputimage);
         if(!f)
         {
             std::cerr << "Could not create output input." << std::endl;
@@ -318,7 +318,6 @@
         f->open(outputimage, spec);
         f->write_image(OIIO::TypeDesc::FLOAT, &img[0]);
         f->close();
-        delete f;
     }
     catch(...)
     {
diff -Nru a/src/apps/ociodisplay/main.cpp b/src/apps/ociodisplay/main.cpp
--- a/src/apps/ociodisplay/main.cpp	2018-12-25 00:02:22.987794023 +0300
+++ b/src/apps/ociodisplay/main.cpp	2018-12-25 00:17:07.401094274 +0300
@@ -95,7 +95,7 @@
         std::cout << "loading: " << filename << std::endl;
         try
         {
-            OIIO::ImageInput* f = OIIO::ImageInput::create(filename);
+            OIIO::ImageInput::unique_ptr f = OIIO::ImageInput::create(filename);
             if(!f)
             {
                 std::cerr << "Could not create image input." << std::endl;
@@ -120,13 +120,8 @@
             memset(&img[0], 0, texWidth*texHeight*components*sizeof(float));
 
             f->read_image(
-#if (OIIO_VERSION >= 10800)
                 OIIO::TypeFloat, 
-#else
-                OIIO::TypeDesc::TypeFloat, 
-#endif
                 &img[0]);
-            OIIO::ImageInput::destroy(f);
         }
         catch(...)
         {
diff -Nru a/src/apps/ociolutimage/main.cpp b/src/apps/ociolutimage/main.cpp
--- a/src/apps/ociolutimage/main.cpp	2018-01-05 04:38:27.000000000 +0300
+++ b/src/apps/ociolutimage/main.cpp	2018-12-25 00:45:56.584363491 +0300
@@ -116,7 +116,7 @@
         processor->apply(imgdesc);
     }
 
-    OIIO::ImageOutput* f = OIIO::ImageOutput::create(outputfile);
+    OIIO::ImageOutput::unique_ptr f = OIIO::ImageOutput::create(outputfile);
     if(!f)
     {
         throw Exception( "Could not create output image.");
@@ -128,7 +128,6 @@
     f->open(outputfile, spec);
     f->write_image(OIIO::TypeDesc::FLOAT, &img[0]);
     f->close();
-    delete f;
 }
 
 
@@ -137,7 +136,7 @@
              const std::string & outputfile)
 {
     // Read the image
-    OIIO::ImageInput* f = OIIO::ImageInput::create(inputfile);
+    OIIO::ImageInput::unique_ptr f = OIIO::ImageInput::create(inputfile);
     if(!f)
     {
         throw Exception("Could not create input image.");
@@ -183,7 +182,7 @@
     std::vector<float> img;
     img.resize(spec.width*spec.height*spec.nchannels, 0);
     f->read_image(OIIO::TypeDesc::TypeFloat, &img[0]);
-    delete f;
+    f->close();
     
     // Repack into rgb
     // Convert the RGB[...] image to an RGB image, in place.
