--- a/cmake/OpenCVCompilerOptions.cmake
+++ b/cmake/OpenCVCompilerOptions.cmake
@@ -98,8 +98,8 @@
   add_extra_compiler_option(-Wformat)
   add_extra_compiler_option(-Werror=format-security -Wformat)
   add_extra_compiler_option(-Wmissing-declarations)
-  add_extra_compiler_option(-Wmissing-prototypes)
-  add_extra_compiler_option(-Wstrict-prototypes)
+  # add_extra_compiler_option(-Wmissing-prototypes)
+  # add_extra_compiler_option(-Wstrict-prototypes)
   add_extra_compiler_option(-Wundef)
   add_extra_compiler_option(-Winit-self)
   add_extra_compiler_option(-Wpointer-arith)
@@ -122,7 +122,7 @@
     add_extra_compiler_option(-Wstrict-aliasing=2)
   else()
     add_extra_compiler_option(-Wno-delete-non-virtual-dtor)
-    add_extra_compiler_option(-Wno-unnamed-type-template-args)
+    # add_extra_compiler_option(-Wno-unnamed-type-template-args)
     add_extra_compiler_option(-Wno-comment)
     if(NOT OPENCV_SKIP_IMPLICIT_FALLTHROUGH
         AND NOT " ${CMAKE_CXX_FLAGS} ${OPENCV_EXTRA_FLAGS} ${OPENCV_EXTRA_CXX_FLAGS}" MATCHES "implicit-fallthrough"
@@ -154,7 +154,7 @@
   endif()
 
   if(OPENCV_WARNINGS_ARE_ERRORS)
-    add_extra_compiler_option(-Werror)
+    add_extra_compiler_option(-Werror -Wno-error=format-truncation -Wno-error=strict-aliasing -Wno-error=array-bounds)
   endif()
 
   if(APPLE)
