--- a/modules/core/test/test_intrin.cpp
+++ b//modules/core/testtest_intrin.cpp
@@ -97,11 +97,13 @@
 DEFINE_SIMD_TESTS(128, AVX512_SKX)
 #endif
 
+#if CV_FP16
 TEST(hal_intrin128, float16x8_FP16)
 {
     CV_CPU_CALL_FP16_(test_hal_intrin_float16, ());
     throw SkipTestException("Unsupported hardware: FP16 is not available");
 }
+#endif
 
 } // namespace intrin128
 
@@ -124,12 +126,14 @@
 DEFINE_SIMD_TESTS(256, AVX512_SKX)
 #endif
 
+#if CV_FP16
 TEST(hal_intrin256, float16x16_FP16)
 {
     //CV_CPU_CALL_FP16_(test_hal_intrin_float16, ());
     CV_CPU_CALL_AVX2_(test_hal_intrin_float16, ());
     throw SkipTestException("Unsupported hardware: FP16 is not available");
 }
+#endif
 
 
 } // namespace intrin256
@@ -140,11 +144,13 @@
     DEFINE_SIMD_TESTS(512, AVX512_SKX)
 #endif
 
+#if CV_FP16
 TEST(hal_intrin512, float16x32_FP16)
 {
     CV_CPU_CALL_AVX512_SKX_(test_hal_intrin_float16, ());
     throw SkipTestException("Unsupported hardware: FP16 is not available");
 }
+#endif
 
 
 } // namespace intrin512
--- a/modules/core/test/test_intrin_utils.hpp
+++ b/modules/core/test/test_intrin_utils.hpp
@@ -18,7 +18,9 @@
 void test_hal_intrin_float32();
 void test_hal_intrin_float64();
 
+#if CV_FP16
 void test_hal_intrin_float16();
+#endif
 
 #ifndef CV_CPU_OPTIMIZATION_DECLARATIONS_ONLY
 
